src/java.base/share/classes/sun/net/InetAddressCachePolicy.java

```diff
@@ -121,40 +113,61 @@ public Integer run() {
                 cachePolicy = DEFAULT_POSITIVE;
             }
         }
-        tmp = java.security.AccessController.doPrivileged (
-          new PrivilegedAction<Integer>() {
-            public Integer run() {
-                try {
-                    String tmpString = Security.getProperty(negativeCachePolicyProp);
-                    if (tmpString != null) {
-                        return Integer.valueOf(tmpString);
-                    }
-                } catch (NumberFormatException ignored) {
-                    // Ignore
-                }
-
-                try {
-                    String tmpString = System.getProperty(negativeCachePolicyPropFallback);
-                    if (tmpString != null) {
-                        return Integer.decode(tmpString);
-                    }
-                } catch (NumberFormatException ignored) {
-                    // Ignore
-                }
-                return null;
-            }
-          });
+        tmp = getProperty(negativeCachePolicyProp,
+                          negativeCachePolicyPropFallback);
 
         if (tmp != null) {
             negativeCachePolicy = tmp < 0 ? FOREVER : tmp;
             propertyNegativeSet = true;
         }
+        long max = TimeUnit.DAYS.toSeconds(7);
+        if (cachePolicy > 0 && cachePolicy < max) {
+            tmp = getProperty(cacheStalePolicyProp,
+                              cacheStalePolicyPropFallback);
+            if (tmp != null) {
+                staleCachePolicy = (int) Math.max(tmp, max);
+            }
+        }
```

Comment:
It is come from the recommendation in the "RFC 8767" see the notion about "cap of 7 days":
https://www.rfc-editor.org/rfc/rfc8767
Depending on the final implementation I can document this, or delete it in the code.

Reply:
From what I can see RFC8767 recommends a cap of 7 days for the TTL itself - that is a maximum value for `cachePolicy` rather than `staleCachePolicy`. It seems to say the the stale timer should be configurable with the text: "The suggested value is between 1 and 3 days".  Personally, I would drop this part of the change. Such suggestions can be adopted (or not) by end users in my opinion.

Reply:
Sounds good, I'll update the patch.